classes/data-stores/ActionScheduler_DBStore.php

```diff
@@ -37,21 +37,47 @@ public function init() {
 	}
 
 	/**
-	 * Save an action.
+	 * Save an action, checks if this is a unique action before actually saving.
+	 *
+	 * @param ActionScheduler_Action $action         Action object.
+	 * @param \DateTime              $scheduled_date Optional schedule date. Default null.
+	 *
+	 * @return int                  Action ID.
+	 * @throws RuntimeException     Throws exception when saving the action fails.
+	 */
+	public function save_unique_action( ActionScheduler_Action $action, \DateTime $scheduled_date = null ) {
```

Comment:
Sorry for not asking this earlier ... but should this be stubbed out in the [abstract class](https://github.com/woocommerce/action-scheduler/blob/3.4.2/classes/abstracts/ActionScheduler_Store.php) as a concrete method that is either no-op, or that throws a doing-it-wrong ('missing implementation error'), or that provides a universal but non-atomic approximation of this operation?

Reply:
Actually adding a stub in the base class will cause issues with the check defined here: https://github.com/woocommerce/action-scheduler/pull/831/files#diff-ba3b93f44933cf6c97c53e92cdc95e3b0e492de26ce0c87430f2d3e832bc95c1R259 which will then evaluate to true for PostStore (and any other custom store). Maybe we can an API which the custom store can use to declare unique support (although in a different PR) and keep this one as it is? wdyt?

Reply:
I saw that check ... then forgot it was there. I think let's leave it as is;  a more formal system for declaring support seems like overkill (to me).